WooCommerce/WooCommerceTests/POS/Presentation/Barcode Scanning/MockTimeProvider.swift

```diff
@@ -1,8 +1,32 @@
 import Foundation
 @testable import WooCommerce
 
+final class MockTimer: Timer {
+    var isCancelled = false
+    let mockTimeProvider: MockTimeProvider
```

Comment:
nit: could there be a retain cycle between `MockTimeProvider` and `MockTimer` since they hold on to each other? Not sure if it's worth keeping one as weak var.

Reply:
Good observation, it would create retain cycles. We can make it weak to avoid some issues in tests.